compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1816,6 +1811,63 @@ public void visit(BLangInvocation iExpr) {
         }
     }
 
+    public void visit(BLangInvocation.BLangActionInvocation aInv) {
+        // For an action invocation, this will only be satisfied when it's an async call of a function.
+        // e.g., start foo();
+        if (aInv.expr == null) {
+            checkFunctionInvocationExpr(aInv);
+            return;
+        }
+
+        // Module aliases cannot be used with remote method call actions
+        if (!validateModuleAliasUsage(aInv)) {
+            return;
+        }
+
+        // Find the variable reference expression type
+        BType exprType = checkExpr(aInv.expr, this.env, symTable.noType);
+        BType actualType = symTable.semanticError;
+        BLangVariableReference varRef = (BLangVariableReference) aInv.expr;
+
+        if (exprType == symTable.semanticError || exprType.tag != TypeTags.OBJECT ||
+                ((varRef.symbol.tag & SymTag.ENDPOINT) != SymTag.ENDPOINT) && !aInv.async) {
```

Comment:
WDYT of using a method instead of all these. Looks a bit too verbose for this method for me :)

Reply:
Just the `(varRef.symbol.tag & SymTag.ENDPOINT) != SymTag.ENDPOINT)` part or the whole thing? The expression as a whole, checks whether it's a remote method call or an async method call.

Reply:
On second thoughts, that's probably not the correct way to check for a remote method call